packages/react/src/runtime/style-cache.tsx

```diff
@@ -11,7 +11,7 @@ import type { ProviderComponent, UseCacheHook } from './types';
  */
 const Cache: any = isServerEnvironment() ? createContext<Record<string, true> | null>(null) : {};
 
-if (!isServerEnvironment()) {
+if (!isServerEnvironment() && typeof document !== 'undefined') {
```

Comment:
is `document` undefined in SSR ?

Reply:
It shouldn't be, no, but I think this is more of a `jsdom` issue.  Maybe the _real fix_ is disabling this: https://github.com/atlassian-labs/compiled/blob/9a15e7420c436ff959e67004b7c6f46bec08f606/packages/react/tsconfig.browser.json#L9 — but I think that increases _more_ bundle size.

tl;dr: We use `document.querySelector` below this and when `dist/browser/runtime/style-cache.js` is accessed by jsdom when it's acting like a browser in our SSR test (I can't tell you why or when it accesses that distribution as it uses `/cjs/` usually) this goes boom.